### PR TITLE
Update knip-reporter.yml

### PR DESCRIPTION
**Pull request by 0x4007**
Noticed this here https://github.com/0x4007/typescript-action/pull/3/files#diff-e300da0383d00d275266a83902a2aff8c7f62da696de4bc47ca5810e69c01ffeR9

The context is that I am building a brand new plugin template, and I started by:
1. Using GitHub's official "[TypeScript Action Starter Template](https://github.com/actions/typescript-action)"
2. Pulling in our [ts-template](https://github.com/ubiquity/ts-template)
3. Finally pulling in our [plugin-template](https://github.com/ubiquity-os/plugin-template)

Is this required?

### CODE REVIEW

**Comment by rndquu** [comment]
> Noticed this here https://github.com/0x4007/typescript-action/pull/3/files#diff-e300da0383d00d275266a83902a2aff8c7f62da696de4bc47ca5810e69c01ffeR9
> 
> The context is that I am building a brand new plugin template, and I started by:
> 
> 1. Using GitHub's official "[TypeScript Action Starter Template](https://github.com/actions/typescript-action)"
> 2. Pulling in our [ts-template](https://github.com/ubiquity/ts-template)
> 3. Finally pulling in our [plugin-template](https://github.com/ubiquity-os/plugin-template)
> 
> Is this required?

TLDR: This PR should be closed.

I don't think `permissions: write-all` is required.

Check [knip-reporter.yml](https://github.com/ubiquity/pay.ubq.fi/blob/45b345aa4893f3e163da4d8ff680e1dc982f193a/.github/workflows/knip-reporter.yml) from the https://github.com/ubiquity/pay.ubq.fi repository which [works](https://github.com/ubiquity/pay.ubq.fi/pull/321#issuecomment-2384556064) fine without `permissions: write-all`. 

Update: Yes, `permissions: write-all` is not required. By default [knip-reporter.yml](https://github.com/ubiquity/ts-template/blob/949a54ce68be95ceff76e457f2d2857ac9203de9/.github/workflows/knip-reporter.yml) runs in a privileged context because of `on: workflow_run`. [Here](https://github.com/ubiquity/ts-template/blob/949a54ce68be95ceff76e457f2d2857ac9203de9/.github/workflows/knip-reporter.yml#L38) github token of the [github-actions](https://github.com/apps/github-actions) app is passed which by [default](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) has `read/write` permission for all of the scopes.

Screenshot 2024-10-01 at 10 18 55
